frontend/packages/dev-console/src/components/topology/TopologyDataController.tsx

```diff
@@ -53,16 +53,58 @@ const Controller: React.FC<ControllerProps> = ({
   render,
   application,
   cheURL,
+  namespace,
   resources,
   loaded,
   loadError,
   utils,
   serviceBinding,
   topologyFilters,
   trafficData,
-  helmResourcesMap,
-}) =>
-  render({
+}) => {
+  const secretCount = React.useRef<number>(0);
+  const helmResourcesMap = React.useRef<HelmReleaseResourcesMap>(null);
+
+  React.useEffect(() => {
+    const count = resources?.secrets?.data?.length || 0;
+    if (count !== secretCount.current) {
+      secretCount.current = count;
+
+      coFetchJSON(`/api/helm/releases?ns=${namespace}`)
+        .then((releases) => {
+          helmResourcesMap.current = releases.reduce((acc, release) => {
+            try {
+              const manifestResources: K8sResourceKind[] = safeLoadAll(release.manifest);
+
+              manifestResources.forEach((resource) => {
+                const resourceKindName = getHelmReleaseKey(resource);
+                if (!acc.hasOwnProperty(resourceKindName)) {
+                  acc[resourceKindName] = {
+                    releaseName: release.name,
+                    chartIcon: release.chart.metadata.icon,
+                    manifestResources,
+                  };
+                }
+              });
+            } catch (e) {
+              // eslint-disable-next-line no-console
+              console.error(e);
+            }
+
+            return acc;
+          }, {});
+        })
+        .catch(() => {
+          helmResourcesMap.current = {};
+        });
+    }
+  }, [namespace, resources, resources.secrets, secretCount]);
+
+  if (!helmResourcesMap.current) {
+    return <LoadingBox />;
+  }
```

Comment:
Maybe instead you want to overload the existing result?
`loaded: helmResourceMap.current && loaded`